PlayChat/AppDelegate.swift

```diff
@@ -14,208 +14,216 @@
-    let channels = dict["Channels"] as! String
-    let chanArray = channels.componentsSeparatedByString(",")
-    maxMessages = dict["MaxMessages"] as! UInt
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions
+    launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+
+    if let path = Bundle.main.path(forResource: "Info", ofType: "plist"),
+       let dictionary = NSDictionary(contentsOfFile: path) as? [String: AnyObject] {
+      chanArray = dictionary["Channels"]!.components(separatedBy: ",")
+      maxMessages = dictionary["MaxMessages"]! as? UInt
+    } else {
+      print("Couldn't read 'Info.plist' file")
+    }
 
     storyboard = UIStoryboard(name: "Main", bundle: nil)
     navigationController = storyboard!.instantiateInitialViewController()
       as? UINavigationController
+
     tabBarController = self.storyboard!
-      .instantiateViewControllerWithIdentifier("TabBarController")
-      as! UITabBarController
-    msgViewController = MessageViewController(maxMessages: maxMessages)
+        .instantiateViewController(withIdentifier: "TabBarController")
+        as? UITabBarController
+    msgViewController = MessageViewController(maxMessages: maxMessages!)
     tabBarController?.delegate = msgViewController
-    
+
     tabBarController?.viewControllers = [buildChannelView(chanArray[0])]
-    for i in 1...chanArray.count-1 {
+    for i in 1...chanArray.count - 1 {
       tabBarController?.viewControllers?.append(buildChannelView(chanArray[i]))
     }
-    
-    FIRApp.configure()
-    GIDSignIn.sharedInstance().clientID = FIRApp.defaultApp()?.options.clientID
+
+    FirebaseApp.configure()
+
+    GIDSignIn.sharedInstance().clientID = FirebaseApp.app()?.options.clientID
     GIDSignIn.sharedInstance().delegate = self
-    
+
     return true
   }
-  
-  func buildChannelView(title : String) -> UIViewController {
+
+  func buildChannelView(_ title: String) -> UIViewController {
     let channelView = UIViewController()
-    let fontBold:UIFont = UIFont(name: "HelveticaNeue-Bold", size: 14)!
+    let fontBold: UIFont = UIFont(name: "HelveticaNeue-Bold", size: 14)!
     channelView.tabBarItem.title = title
     channelView.tabBarItem.setTitleTextAttributes(
       [
-        NSForegroundColorAttributeName: UIColor.grayColor(),
-        NSFontAttributeName: fontBold
-      ], forState: UIControlState.Normal)
+        NSAttributedStringKey.foregroundColor: UIColor.gray,
+        NSAttributedStringKey.font: fontBold
+      ], for: UIControlState())
     channelView.tabBarItem.setTitleTextAttributes(
       [
-        NSForegroundColorAttributeName: UIColor.blackColor(),
-        NSFontAttributeName: fontBold
-      ], forState: UIControlState.Selected)
-    
-    let tableView:UITableView = UITableView()
+        NSAttributedStringKey.foregroundColor: UIColor.black,
+        NSAttributedStringKey.font: fontBold
+      ], for: UIControlState.selected)
```

Comment:
You can omit the type `UIControlState` here and just let the compiler infer the type, since it's specified in the method signature and thus unambiguous.

Reply:
Done.